k8s/apps/statefulsets.go

```diff
@@ -328,3 +330,33 @@ func (c *Client) validatePersistentVolumeClaim(pvc *corev1.PersistentVolumeClaim
 	}
 	return nil
 }
+
+// DeleteStatefulSetPods deletes pods for the given statefulset name and namespace
+func (c *Client) DeleteStatefulSetPods(name, namespace string, timeout time.Duration) error {
+	sset, err := c.GetStatefulSet(name, namespace)
+	if err != nil {
+		return err
+	}
+
+	pods, err := c.GetStatefulSetPods(sset)
+	if err != nil {
+		return err
+	}
+
+	var podsNamesToDelete []string
+	var podsToDelete []corev1.Pod
+	for _, pod := range pods {
+		podsNamesToDelete = append(podsNamesToDelete, pod.Name)
+		podsToDelete = append(podsToDelete, pod)
```

Comment:
Why can't we just use the list `pods` everywhere? Why do we need to construct this new list? It seems you are already using `pods` on DeletePods function but using `podsToDelete` in WaitForPodsToBeDeleted.

Reply:
Done, removed it